soc/nordic/nrf54l/Kconfig

```diff
@@ -80,6 +80,9 @@ config SOC_NRF_FORCE_CONSTLAT
 	  and predictable latency will be at the cost of having increased power consumption.
 
 config SOC_NRF54L_NORMAL_VOLTAGE_MODE
+	select DEPRECATED
+	help
+	  This mode is deprecated and shall no longer be used.
```

Comment:
also, if deprecating functionality should be kept, and soc.c code is deleted

Reply:
> is it worth deprecating on socs in the engineering phase?

Having this option enabled can be actually harmful when MDK 8.67.0 (nrfx 3.7.0) is used, so let's remove it instead of deprecating